### ISSUE TITLE
Cannot process argument transformation on parameter 'Skip'

### ISSUE THREAD

**Issue by hullscotty1986** [rmbolger/Posh-ACME#389]
Getting error:
Line |
  94 |  Import-PAConfig
     |  ~~~~~~~~~~~~~~~
     | Cannot process argument transformation on parameter 'Skip'.
     | Cannot convert value "System.Object[]" to type
     | "System.Boolean". Boolean parameters accept only Boolean
     | values and numbers, such as $True, $False, 1 or 0.

On version 4.7+

**Comment by rmbolger** [comment]
Hey @hullscotty1986. Sorry to hear you're having issues. Can you verify the specific module version you're on? Also are you sure it was line 94 and not line 64? I'm not seeing any code that would be calling something with a `Skip` parameter around line 94.

Line 64 calls an internal function related to controlling cert validation that has a `Skip` parameter though. The error you're getting implies the config file for the ACME server you're using has become corrupted in some way. It would help if you can post the content of it here (there's nothing sensitive in it). The exact path is going to depend on your username and the ACME server you're using. Here's an easy way to open Explorer to the config root though:

```powershell
start "$($env:LOCALAPPDATA)\Posh-ACME"
```

You'll want to open the sub-folder for the ACME server you're using and then open `dir.json` and post the content here.

**Comment by hullscotty1986** [comment]
I'm not sure how I did it, but I had a LE_PROD folder as well as the acme-v02.api.letsencrypt.org folder.
I deleted the LE_PROD folder as it didn't have anything useful in it, and that seems to have fixed the issue. I don't know if its a dodgy config file? If you want me to, I can PM you a Zip of my structure to recreate

**Comment by hullscotty1986** [comment]
I was using 4.9 (latest) and the dir.json that I deleted is:

{
    "location":  "https://acme-v02.api.letsencrypt.org/directory",
    "DisableTelemetry":  false,
    "SkipCertificateCheck":  false,
    "newAccount":  "https://acme-v02.api.letsencrypt.org/acme/new-acct",
    "newOrder":  "https://acme-v02.api.letsencrypt.org/acme/new-order",
    "newNonce":  "https://acme-v02.api.letsencrypt.org/acme/new-nonce",
    "keyChange":  "https://acme-v02.api.letsencrypt.org/acme/key-change",
    "revokeCert":  "https://acme-v02.api.letsencrypt.org/acme/revoke-cert",
    "meta":  {
                 "caaIdentities":  [
                                       "letsencrypt.org"
                                   ],
                 "termsOfService":  "https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf",
                 "website":  "https://letsencrypt.org"
             },
    "nonce":  "0102-GKe-m5K3JtaTWJIdXmZHG7D14dFQOZ6NlkR27-NCoU"
}

**Comment by rmbolger** [comment]
Weird. I wonder if I've got (or had) a bug that is not properly recognizing an existing server folder when using `Set-PAServer` and ends up creating a duplicate with the new default name (which uses the alias). The config file is fine and it was definitely the dupe that was fouling things up.

There's currently an assumption in the code that each server folder has a `dir.json` with a unique `location` field. So when it got two results, it essentially sent an array of bools to that function instead of a single one. I wish I knew how the dupe got created though. I'll try to play around with some possibilities on my end. But you should be all fixed.

**Comment by hullscotty1986** [comment]
Yeah that will be it, I just looked at both dir.jsons and the locations are the same. It was years ago I set this up and it wasn't an issue until it upgraded to 4.7 however I have only just noticed it because I got the emails from LetsEncrypt saying the certs were about to expire.

What's even weirder is that if I don't use -Force on Import-Module it works ok, but then when I do 
Set-PAServer -DirectoryUrl LE_PROD
I get 
Set-PAServer : Cannot convert 'System.Object[]' to the type 'System.Uri' required by parameter 'Uri'. Specified method is not supported.
At line:14 char:1
+ Set-PAServer -DirectoryUrl LE_PROD
Hopefully no one else would have had this issue!

**Comment by rmbolger** [comment]
If you don't use `-Force` and the module is already loaded, it effectively doesn't do anything which is why it might seem to work. It just uses the already loaded copy.
